### ISSUE TITLE
Seams are inconsistent when painted, do not aligned properly, and change when multiple (cloned) parts are added to bed.

### ISSUE THREAD

**Issue by Vantage-DS** [SoftFever/OrcaSlicer#5621]
### Is there an existing issue for this problem?

- [X] I have searched the existing issues

### OrcaSlicer Version

2.1.0 (all versions)

### Operating System (OS)

macOS, Windows

### OS Version

N/A

### Additional system information

N/A

### Printer

P1S - Though, irrelevant

### How to reproduce

1) Slice model.
2) Seams are not adequately placed for cosmetic look of final print.
3) Paint seams manually, as nearest/aligned or other options do not give good enough results.

### Actual results

Seams are notoriously difficult to place manually on objects that do not have perfectly straight edges.
Results of seams, even after painting, do not line up or properly cling to sharp edges when using "nearest", nor do they align neatly when using "aligned".

Seam painting takes a very long time on some of my models. One example where i blocked seams everywhere on the model to try get neat results:
<img width="1019" alt="SeamPainted(single)" src="https://github.com/SoftFever/OrcaSlicer/assets/42471125/e198f300-7b60-4be0-be95-a4ea710d0de1">

The result is still random:
<img width="1113" alt="SeamResult(single)" src="https://github.com/SoftFever/OrcaSlicer/assets/42471125/2b437100-f70d-43d3-802b-c631250057c6">

Blocking and repainting the tiniest parts of a seam line are the only way to fix this:
Seam close up-
<img width="829" alt="Screenshot 2024-06-07 at 15 10 53" src="https://github.com/SoftFever/OrcaSlicer/assets/42471125/57efe9b8-a78c-4dac-98e4-c1adb671bd03">

Fully zoomed in, the painted view of ^that seam painting looks like-
<img width="925" alt="Screenshot 2024-06-07 at 15 11 15" src="https://github.com/SoftFever/OrcaSlicer/assets/42471125/2504f0ad-f249-4847-a98b-59455653d293">

The minute changes i have to make to make the seam align properly-
<img width="883" alt="Screenshot 2024-06-07 at 15 12 23" src="https://github.com/SoftFever/OrcaSlicer/assets/42471125/da433968-df31-444c-9111-d85ab21f82de">

The new result of ^that minute change-
<img width="823" alt="Screenshot 2024-06-07 at 15 12 33" src="https://github.com/SoftFever/OrcaSlicer/assets/42471125/a9707991-7def-4f17-8e03-699e826d60f9">

The problem is further exaggerated when cloning a part.

The original seam painting (same screenshot from earlier):

<img width="909" alt="SeamPatined(multi)" src="https://github.com/SoftFever/OrcaSlicer/assets/42471125/bf88a0e0-fedb-490f-b388-9ad7eed76f10">

The actual seam when 2 objects are now placed on the bed:

Not only does the seam layout from the 'single' model change it's location, which means when i have 2 clones on a plate, the seam needs refining yet again, the seams are also being put where they are supposed to be blocked-
<img width="912" alt="SeamResult(multi)" src="https://github.com/SoftFever/OrcaSlicer/assets/42471125/3446b49e-deb8-4a3c-a1c8-43c4acdc6fa3">

### Expected results

An aligned/nearest seam should cling to an edge in a straight and neat line like this:
<img width="230" alt="Screenshot 2024-06-07 at 15 07 24" src="https://github.com/SoftFever/OrcaSlicer/assets/42471125/dbfc71b7-30bc-4f01-83ad-1455736f9c97">

The seam should stay in that location regardless of how many objects are on the build plate.

The computation of nearest or aligned needs be refined to actually align.

OrcaSlicer is brilliant, and this has been a problem way before OrcaSlicer existed, but i think it's time seams were fixed before i lose my sanity 🙃

### Project file & Debug log uploads

n/a

### Checklist of files to include

- [ ] Log file
- [ ] Project file

**Comment by Vantage-DS** [comment]
More problems....

Switching from my P1S to a Prusa MK4, using near identical slicing settings (other than speed).


This tiny blue dot is where it want's to put the seam (left side of picture)-
<img width="927" alt="Screenshot 2024-06-10 at 18 00 46" src="https://github.com/SoftFever/OrcaSlicer/assets/42471125/8466408e-090c-4c10-807e-0715779173f3">

This is the painting. There is no reason for it-
<img width="880" alt="Screenshot 2024-06-10 at 18 01 00" src="https://github.com/SoftFever/OrcaSlicer/assets/42471125/c848c277-f7b4-4918-9b32-1066eef44427">


I try go overboard, and give the slicer more than enough room to add a seam to the correct area, just for testing-
<img width="900" alt="Screenshot 2024-06-10 at 18 01 31" src="https://github.com/SoftFever/OrcaSlicer/assets/42471125/9ba1c525-6d4d-4b49-854e-5c63cef79ec4">


This is the result. The blue dot/seam still wants to be on the left-
<img width="969" alt="Screenshot 2024-06-10 at 18 01 21" src="https://github.com/SoftFever/OrcaSlicer/assets/42471125/1a4572c4-657a-4b84-93f1-962ab9f0cfbe">



I've got 3 sizes of this object, and each time i need to slice a variation (1 on the bed, 2 on the bed, 4 on the bed, or even now, a different printer) the seams need refining. It's taking so much time and i have no idea why it's being like this.

**Comment by NVNDO** [comment]
Seams in OS and BS are horrible. I don't understand why its so hard to implement this properly like in Cura. 

90% of time I have to draw the seam myself and not even then, like mentioned here it as it should, as the software randomly bends or cut the seam for no reason.

**Comment by Vantage-DS** [comment]
Yeah, Cura has great seam placement. I've never painted seams on Cura, because I never needed to 😄

**Comment by DARKFiB3R** [comment]
Bump

**Comment by adripo** [comment]
Hi everyone, I'm still running into the seam alignment issues mentioned here. I’d be happy to help. whether that's testing patches or diving into the code. Could someone point out which part of the logic might need extra attention? I'd appreciate any guidance so we can get this resolved together.

**Comment by NVNDO** [comment]
Well.. that I call showing initiative @adripo - I hope when can provide the necessary information and you can sort this out. It would be a game-changer! 🙏🏼

**Comment by kbroasterc** [comment]
This and curved seams when set to back has me pulling my hair out, I love to have a fix for this

**Comment by silvernull** [comment]
This issue constantly plagues me also.

![Image](https://github.com/user-attachments/assets/cb585e57-5ced-4693-9cb2-581e239f3b2c)

**Comment by Vantage-DS** [comment]
Simple straight line - Both Aligned and Near:

<img width="1214" alt="Image" src="https://github.com/user-attachments/assets/e90e16f1-3774-40fc-b83d-c6769102247c" />
<img width="1271" alt="Image" src="https://github.com/user-attachments/assets/16c5eed4-9088-4c8c-85db-4dde3a394c7f" />
<img width="1292" alt="Image" src="https://github.com/user-attachments/assets/26fa5fd2-fc03-409e-aebf-1a7584848e9b" />



There must be some code from Prusa Slicer that can be pulled.

I drew some really lazy seams in Prusa (about 10 seconds of work) and it lined up perfectly..... The alternative in Orca Slicer took me about 20 minutes to fine tune, and even then it wasn't guaranteed.

**Comment by johyphenel** [comment]
+1, trying to paint a seam to be aligned and in a corner feels impossible with Orca.

**Comment by johyphenel** [comment]
Bump to keep open.

**Comment by Vantage-DS** [comment]
Had fun again with this in the latest alpha, so it's still a thing.

Constant fine tuning - blocking and repainting the tiniest sub millimetre seam placements, just to get them to go in a straight line.

**Comment by dandudikof** [comment]
Same problems, here , with painted and back seams , unable to get consistent straight back
or painted where painted.

I started 3d printing with Cura, then Orca and now due to basics like this not working in Orca, trying Prusa, which seems to have essentials working before implementing latest and greatest 
(which are very nice), but not when basics are not working!

**Comment by DImagine** [comment]
@Vantage-DS, @NVNDO, @DARKFiB3R, @adripo, @kbroasterc, @silvernull, @johyphenel, @dandudikof

There is an open pull request (#12974) that offers an alternative approach to the seam placement problem discussed here. It adds modifier-object-based seam placement — either by pinpointing an exact location or by marking allow/forbid zones. Because placement is geometry-based rather than surface-painted, seam positions stay consistent across cloned parts and reslicing.

I'd really appreciate it if you could test this feature on real-world models and share your feedback on the pull request.
